collection.spec.js

```diff
@@ -1,143 +1,152 @@
-const {Collection} = require('./index');
+const { Collection } = require("./index");
+const get = require("lodash/get");
 
 function getClient({
-  getCollectionBySlug = (slug) => {
+  getCollectionBySlug = slug => {
     return Promise.resolve({
-      slug: 'home',
+      slug: "home",
       summary: null,
       items: [
         {
-          'associated-metadata': {layout: 'ArrowFourColTwelveStories'},
-          type: 'collection',
-          slug: 'arrow-collection',
-        },
-      ],
+          "associated-metadata": { layout: "ArrowFourColTwelveStories" },
+          type: "collection",
+          slug: "arrow-collection"
+        }
+      ]
     });
   },
 
-  getInBulk = (request) => {
+  getInBulk = request => {
     let collectionItem = {};
     const requestKeys = Object.keys(request.requests);
 
     if (requestKeys.length > 0) {
-      requestKeys.map((req) => {
+      requestKeys.map(req => {
         let items = [];
         const limit = request.requests[req].limit || 40;
 
         for (let i = 0; i < limit; i++) {
           items.push({
-            'associated-metadata': {},
-            type: 'collection',
-            slug: `football-sports-${i}`,
+            "associated-metadata": {},
+            type: "collection",
+            slug: `football-sports-${i}`
           });
         }
 
         Object.assign(collectionItem, {
           [req]: {
             slug: req,
-            items: items,
-          },
+            items: items
+          }
         });
       });
     }
 
     return Promise.resolve({
-      results: collectionItem,
+      results: collectionItem
     });
-  },
+  }
 } = {}) {
   const clientObj = {
     getCollectionBySlug,
-    getInBulk,
+    getInBulk
   };
   return clientObj;
 }
 
-describe('Collection', function () {
-  describe('Returns Home Collection based on defaultNestedLimit', function () {
-    it('Returns home-collection with depth of 1 and defaultNestedLimit of 3', async function () {
+function request(slug, params) {
+  throw { response: { status: 404 } };
+}
+
+function catch404(e, defaultValue) {
+  const statusCode = get(e, ["response", "status"]);
+  if (statusCode === 404) return Promise.resolve(defaultValue);
+  throw e;
+}
+
+function getClientWhenCollectionDoesNotExist({
+  getCollectionBySlug = (slug, params) => {
+    try {
+      return request("/api/v1/collections/" + slug, {
+        qs: params
+      });
+    } catch (e) {
+      return catch404(e, null);
+    }
+  }
+} = {}) {
+  return { getCollectionBySlug };
+}
+
```

Comment:
Instead, you can reuse the `getClient` method above by passing it `getCollectionBySlug` function as a parameter

Reply:
```js
const clientThatCallsCollThatDoesntExist = getClient({
  getCollectionBySlug: () => {
    return Promise.resolve({response: { status: 404 }})
  }
})
```